src/Geometry/globalgeometry.jl

```diff
@@ -146,6 +146,18 @@ function great_circle_distance(
     )
 end
 
+"""
+    euclidean_distance(pt1::XYPoint, pt2::XYPoint)
+
+Compute the 2D or 3D Euclidean distance between `pt1` and `pt2`.
+"""
+function euclidean_distance(
+    pt1::T,
+    pt2::T,
+) where {T <: Union{XYPoint, XZPoint, XYZPoint}}
```

Comment:
we might as well support single dimensions as well:
```suggestion
) where {T <: Union{XPoint, YPoint, ZPoint, XYPoint, XZPoint, XYZPoint}}
```

Reply:
Yes, I had considered that honestly

Reply:
I added that, related tests and squashed commits. Please let me know if it looks good to merge. Thanks!